library/std/src/sys/hermit/stdio.rs

```diff
@@ -118,3 +118,23 @@ pub fn is_ebadf(_err: &io::Error) -> bool {
 pub fn panic_output() -> Option<impl io::Write> {
     Some(Stderr::new())
 }
+
+#[unstable(feature = "is_terminal", issue = "80937")]
+impl io::IsTerminal for sys::stdio::Stdin {
```

Comment:
The types in `sys` are private, so there is no need to use a trait here. These can just be inherent methods on the private type.